src/management-system-v2/components/modeler-share-modal.tsx

```diff
@@ -0,0 +1,146 @@
+import { FC, useState } from 'react';
+import { Modal, Button, Tooltip, Space, Card, Divider, message } from 'antd';
+import {
+  ShareAltOutlined,
+  LinkOutlined,
+  ExportOutlined,
+  LeftOutlined,
+  RightOutlined,
+  CopyOutlined,
+} from '@ant-design/icons';
+import useModelerStateStore from '@/lib/use-modeler-state-store';
+import { copyProcessImage } from '@/lib/process-export/copy-process-image';
+import ModelerShareModalOptionPublicLink from './modeler-share-modal-option-public-link';
+import ModelerShareModalOptionEmdedInWeb from './modeler-share-modal-option-embed-in-web';
+import ModelerShareModalOption from './modeler-share-modal-option';
+
+type ShareModalProps = {
+  onExport: () => void;
+};
+
+const ModelerShareModalButton: FC<ShareModalProps> = ({ onExport }) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [isSharePublicLinkSelected, setIsSharePublicLinkSelected] = useState(false);
```

Comment:
I think you could avoid the specific `is...Selected` state variables since your activeIndex already defines what is selected.

Maybe put something like `subOptions: <ModelerShareModalOption[...] />` into your options and then do somethink like this in the Modal:
```
[...]
</Space>
{activeIndex !== null ? (
  <>
    <Divider style={{ backgroundColor: 'black' }} />
    {options[activeIndex].subOption}
  </>
) : null}
```